kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableSchemaEvolutionSuite.scala

```diff
@@ -1156,6 +1157,36 @@ class DeltaTableSchemaEvolutionSuite extends DeltaTableWriteSuiteBase with Colum
     }
   }
 
+  // TODO: need to remove this once we support schema evolution with clustering columns
+  test("Cannot update schema with clustering columns") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = Table.forPath(engine, tablePath)
+      val initialSchema = new StructType()
+        .add("a", StringType.STRING, true)
+        .add("c", IntegerType.INTEGER, true)
+
+      createEmptyTable(
+        engine,
+        tablePath,
+        initialSchema,
+        clusteringCols = List(new Column("c")),
+        tableProperties = Map(
+          TableConfig.COLUMN_MAPPING_MODE.getKey -> "id",
+          TableConfig.ICEBERG_COMPAT_V2_ENABLED.getKey -> "true"))
+
+      val currentSchema = table.getLatestSnapshot(engine).getSchema
+      val newSchema = new StructType()
+        .add("c", StringType.STRING, true, currentSchema.get("c").getMetadata)
+        .add("a", StringType.STRING, true, currentSchema.get("a").getMetadata)
+
+      assertSchemaEvolutionFails[IllegalArgumentException](
```

Comment:
we are throwing KernelException [here](https://github.com/delta-io/delta/pull/4327/files#diff-5abbcf2158a0779bd57ce93b0041f6feb643c3a0bcba6d211dc0b4c062a67619R373), how come we are expecting illegal argument exception?